jupyterhub/templates/hub/deployment.yaml

```diff
@@ -75,8 +75,8 @@ spec:
           persistentVolumeClaim:
             claimName: {{ include "jupyterhub.hub-pvc.fullname" . }}
         {{- end }}
-      {{- if .Values.rbac.enabled }}
-      serviceAccountName: {{ include "jupyterhub.hub.fullname" . }}
+      {{- if .Values.hub.serviceAccount.create }}
```

Comment:
@consideRatio I would thus amend your Change Request 1 to unconditionnaly set the serviceAccountName. If `rbac.enable: false` and `serviceAccount.create: false` and `serviceAccount.name:` then it will use `default` anyway, which is how it seems to be done in helm boilerplate.

Reply:
I believe this case is already covered by CR1 though? If no name is given, and create=false, it won't reference any serviceAccount, thus just default to whatever the cluster default is, right? If the user wants to run as `default` even if the cluster default is something else, they can specify that in the name. Or am I missing something? There is some nested logic going on here and I'm happy it's no deeper than in is to be honest :sweat_smile:

Reply:
Honestly I would just do the same way as the whole community is doing instead of having specific logic for this Chart only !

Reply:
Closing this as resolved, but asked a followup question about this in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2736/files#r888795016 where we can look at the latest code.